src/Data/List/Properties.agda

```diff
@@ -828,12 +818,65 @@ drop-take-suc-tabulate : ∀ {n} (f : Fin n → A) (i : Fin n) → let m = toℕ
 drop-take-suc-tabulate f i rewrite sym (toℕ-cast (sym (length-tabulate f)) i) | sym (lookup-tabulate f i)
   = drop-take-suc (tabulate f) (cast _ i)
 
+drop-all : (n : ℕ) (xs : List A) → n ≥ length xs → drop n xs ≡ []
+drop-all n       []       _ = drop-[] n
+drop-all (suc n) (x ∷ xs) p = drop-all n xs (≤-pred p)
+
 -- Dropping m elements and then n elements is same as dropping m+n elements
 drop-drop : (m n : ℕ) → (xs : List A) → drop n (drop m xs) ≡ drop (m + n) xs
 drop-drop zero n xs = refl
 drop-drop (suc m) n [] = drop-[] n
 drop-drop (suc m) n (x ∷ xs) = drop-drop m n xs
 
+------------------------------------------------------------------------
+-- replicate
+
+length-replicate : ∀ n {x : A} → length (replicate n x) ≡ n
+length-replicate zero    = refl
+length-replicate (suc n) = cong suc (length-replicate n)
+
+lookup-replicate : ∀ n (x : A) (i : Fin n) → lookup (replicate n x) (cast (sym (length-replicate n)) i) ≡ x
+lookup-replicate (suc n) x zero    = refl
+lookup-replicate (suc n) x (suc i) = lookup-replicate n x i
+
+map-replicate :  ∀ (f : A → B) (x : A) n →
+                 map f (replicate n x) ≡ replicate n (f x)
+map-replicate f x zero    = refl
+map-replicate f x (suc n) = cong (_ ∷_) (map-replicate f x n)
+
+zipWith-replicate : ∀ n (_⊕_ : A → B → C) (x : A) (y : B) →
+                    zipWith _⊕_ (replicate n x) (replicate n y) ≡ replicate n (x ⊕ y)
+zipWith-replicate zero    _⊕_ x y = refl
+zipWith-replicate (suc n) _⊕_ x y = cong (x ⊕ y ∷_) (zipWith-replicate n _⊕_ x y)
+
+------------------------------------------------------------------------
+-- iterate
+
+length-iterate : ∀ n {f} {x : A} → length (iterate f x n) ≡ n
+length-iterate zero    = refl
+length-iterate (suc n) = cong suc (length-iterate n)
+
+iterate-id : ∀ n {x : A} → iterate id x n ≡ replicate n x
+iterate-id zero    = refl
+iterate-id (suc n) = cong (_ ∷_) (iterate-id n)
+
+module _ n f x where
+  open import Data.List.Base
+  private
+    xs = iterate f x n
+    n≥length[xs] : n ≥ length xs
+    n≥length[xs] rewrite length-iterate n {f} {x} = ≤-refl
+    
+  take-iterate : take n xs ≡ xs
+  take-iterate = take-all n xs n≥length[xs]
+
+  drop-iterate : drop n xs ≡ []
+  drop-iterate = drop-all n xs n≥length[xs]
+
+lookup-iterate : ∀ n f (x : A) (i : Fin n) → lookup (iterate f x n) (cast (sym (length-iterate n)) i) ≡ ℕ.iterate f x (toℕ i)
```

Comment:
I think it's a good heuristic to have the parameters to a function proceed from left-to-right from 'what stays fixed' through to 'what varies throughout' (I learnt this from @JacquesCarette ... ;-)), moreover in a way which correlates with the corresponding argument order in the function invocation `iterate f x n`. So apologies are in order for not reflecting that in the module parameter list above... where `f` and `x` should come before `n`.

Indeed, within the scope of a given `f` and `x`, the module can/should consider *all* the proofs about `iterate f x` for various arguments `n`, which may vary internally (or else bind/be bound by the types of arguments like `i`), or which may be fixed as in `take-iterate` etc. 

So a bit of gentle refactoring is in order here, I think, to get the maximum value out of working in a module scope where `f` and `x` are 'fixed'...

Reply:
Thanks @jamesmckinna! I'll try to refactor all the iterate proofs into this module. I've rearranged the arguments in all the proofs.